cradle-core/src/test/java/com/exactpro/cradle/messages/MessageBatchToStoreJoinTest.java

```diff
@@ -193,12 +202,26 @@ private MessageBatchToStore createBatch(BookId bookId, String sessionAlias, long
         return messageBatchToStore;
     }
 
-    private MessageBatchToStore createEmptyBatch() {
+    private static MessageBatchToStore createEmptyBatch() {
         return new MessageBatchToStore(MAX_SIZE);
     }
-    
-    private MessageBatchToStore createFullBySizeBatch(BookId bookId, 
+
+    static MessageBatchToStore createFullBySizeBatch(BookId bookId,
             String sessionAlias, long startSequence, Direction direction, Instant startTimestamp) throws CradleStorageException {
-        return createBatch(bookId, sessionAlias, startSequence, direction, startTimestamp, 1, MAX_SIZE);
+        return createBatch(bookId, sessionAlias, startSequence, direction, startTimestamp, 1,
+                MAX_SIZE - (MessagesSizeCalculator.MESSAGE_BATCH_CONST_VALUE
+                        + MessagesSizeCalculator.MESSAGE_SIZE_CONST_VALUE + MessagesSizeCalculator.MESSAGE_LENGTH_IN_BATCH));
+    }
+
+    private int getRealBatchSize(StoredMessageBatch batch, int bufferSize) throws SerializationException {
```

Comment:
This method looks strange: to get real batch size we need to know in advance the maximum buffer to where it will be stored. Why? Can batch size be calculated without this?